css-module.spec.mjs

```diff
@@ -60,6 +60,7 @@ describe('css-module loader', { concurrency: true }, () => {
 				Qux: 'Qux',
 				Zed: 'Zed',
 				img: 'img',
+				nested: 'nested',
```

Comment:
we could have something like this to represent the nested :

```json
{
    "Foo": "Foo",
    "Bar": "Bar",
    "Qux": "Qux",
    "Zed": "Zed",
    "img": "img",
    "nested": {
        "something": "something"
    }
}
```

but it would change the spec

Reply:
It's an additive change (nesting wasn't supported before), so that's fine. I think your proposal in the comment (`something.nested`) is better as it avoids a risk of collision 🙂

I think the README would need to be updated (and if so, before cutting a release). I can do subsequently if you don't want to handle in this PR, but I can't til I'm back from Helsinki next week (same for cutting the release).

Reply:
I'm going to experiment with the loader with react typescript and tailwind to find out which is the best behaviour to have.

Reply:
Here example of dum component:
```tsx
import { useState } from 'react';
import styles from './basic.module.css';
import type { FC } from 'react';

const Basic: FC = () => {
    const [show, setShow] = useState(false);

    return (
        <div className={styles.container}>
            <button
                className={styles.button}
                onClick={() => setShow(!show)}
            >Toggle</button>
            {show && <p>Content</p>}
        </div>
    );
};

export default Basic;
```

here the test suite:
```tsx
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { render, screen, act } from "@testing-library/react";
import Basic from "./basic.tsx";

describe("Basic", () => {
    it("should render", () => {
        const { container } = render(<Basic />);
        assert.ok(container);
        container.remove();
    });

    it("should toggle content", () => {
        const { unmount } = render(<Basic />);
        const button = screen.getByRole("button");
        const content = screen.queryByText("Content");

        assert.ok(button);
        assert.strictEqual(content, null);

        act(() => {
            button.click();
        });

        assert.ok(screen.queryByText("Content"));
        unmount();
    });

    it("should have correct class names", () => {
        render(<Basic />);
        const container = screen.getByRole("button").parentElement;

        if (!container) {
            throw new Error("Container not found");
        }

        assert.strictEqual(container.className, "container");
        
        const button = container.querySelector("button");

        if (!button) {
            throw new Error("Button not found");
        }

        assert.strictEqual(button.className, "button");
    });
});
```

This is currently failing. That's why I think having nested keys in a js object seems more appropriate.

_I'm well aware that nobody in their right mind would do such a test_.

Reply:
With lasted commit this test pass